src/ui/ResponsiveComponents/ResponsiveRecommendation.ts

```diff
@@ -98,10 +98,10 @@ export class ResponsiveRecommendation implements IResponsiveComponent {
     ResponsiveComponentsUtils.deactivateSmallRecommendation(this.recommendationRoot);
   }
 
-  private buildDropdown(): ResponsiveDropdown {
+  private buildDropdown(dropdownMock: ResponsiveDropdown): ResponsiveDropdown {
     let dropdownContent = this.buildDropdownContent();
     let dropdownHeader = this.buildDropdownHeader();
-    let dropdown = new ResponsiveDropdown(dropdownContent, dropdownHeader, this.coveoRoot);
+    let dropdown = dropdownMock ? dropdownMock : new ResponsiveDropdown(dropdownContent, dropdownHeader, this.coveoRoot);
```

Comment:
It's a bit weird. The method declare the parameter non optional, but the code assume that it can be undefined. Why is that ? Should not be possible, even more so that it's a internal/private method.

Also, rename it so that it does not mention `mock`.